src/datasets/mnist.nim

```diff
@@ -95,19 +97,97 @@ proc read_mnist_labels*(labelsPath: string): Tensor[uint8] {.noInit.}=
   ##     - W, width
   ##
   ## MNIST data can be downloaded here: http://yann.lecun.com/exdb/mnist/
-  ## It must be uncompressed before use.  Download and decompression will be automated in the future.
-  # Compressed reader pending https://github.com/nim-lang/zip/pull/20
-  if not existsFile(labelsPath):
-    raise newException(IOError, "MNIST labels file \"" & labelsPath & "\" does not exist")
+  if not existsFile(imgsPath):
+    raise newException(IOError, "MNIST images file \"" & imgsPath & "\" does not exist")
+
+  let stream = newGzFileStream(imgsPath, mode = fmRead)
```

Comment:
Does newGzFileStream also works for uncompressed files?

If no I'd like to have a `if uncompressed`

Reply:
I think it assumes its compressed:

https://github.com/nim-lang/zip/blob/master/zip/gzipfiles.nim#L61